graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -70,13 +72,15 @@
     public StreamRouterEngine(@Assisted List<Stream> streams,
                               @Assisted ExecutorService executorService,
                               StreamFaultManager streamFaultManager,
-                              StreamMetrics streamMetrics) {
+                              StreamMetrics streamMetrics,
+                              @DefaultStream Stream defaultStream) {
```

Comment:
This needs to be a `Provider<Stream>` to avoid the problems when starting the server.